app/ui-utils/client/lib/RoomHistoryManager.js

```diff
@@ -333,7 +333,18 @@ export const RoomHistoryManager = new class {
 	}
 
 	clear(rid) {
+		const query = { rid };
+		const options = {
+			sort: {
+				ls: -1,
+			},
+			limit: 50,
+		};
+		const retain = ChatMessage.find(query, options).fetch();
```

Comment:
Better to get the `ls` of the 50th item and use it to restrict the query to remove than reinsert everything.

Reply:
Makes sense, will modify this part.

Reply:
Modified this part